src/overlays/actors/ovl_Bg_Tobira01/z_bg_tobira01.c

```diff
@@ -28,14 +28,78 @@ const ActorInit Bg_Tobira01_InitVars = {
     (ActorFunc)BgTobira01_Update,
     (ActorFunc)BgTobira01_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/func_80B12430.s")
+void BgTobira01_Action(BgTobira01* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s16 actorCutscene = this->dyna.actor.csId;
+    s16 tempDoorYPosition;
+
+    if (this->cutscenePlayed != 0) {
+        if (CutsceneManager_GetCurrentCsId() == CS_ID_GLOBAL_TALK) {
+            CutsceneManager_Stop(CS_ID_GLOBAL_TALK);
+        } else if (CutsceneManager_IsNext(actorCutscene)) {
+            CutsceneManager_StartWithPlayerCs(actorCutscene, &this->dyna.actor);
+            SET_WEEKEVENTREG(WEEKEVENTREG_88_40);
+            this->cutscenePlayed = 0;
+        } else {
+            CutsceneManager_Queue(actorCutscene);
+        }
+    } else if (!(CHECK_WEEKEVENTREG(WEEKEVENTREG_88_40)) && (this->doorYPositionTick == 0) &&
+               (play->actorCtx.playerImpact.timer != 0) && (play->actorCtx.playerImpact.type == 0) &&
```

Comment:
```suggestion
               (play->actorCtx.playerImpact.timer != 0) && (play->actorCtx.playerImpact.type == PLAYER_IMPACT_GORON_GROUND_POUND) &&
```
There's an enum for this